contracts/contracts/eternalstorage/EternalStorage.sol

```diff
@@ -0,0 +1,260 @@
+pragma solidity 0.4.23;
```

Comment:
This contract has a lot going on. Did you make this from scratch or does this exist somewhere else?